### PR TITLE
[CLEANUP] Remove _actions functionality

### PR DESCRIPTION
**Pull request by thoov**
Along with related deprecations, asserts, and tests.

Related: [ember-2-legacy PR](https://github.com/emberjs/ember-2-legacy/pull/17)

Apart of: #15876

### CODE REVIEW

**Comment by thoov** [comment]
Tests have been ported to the legacy addon

File: packages/ember-runtime/lib/mixins/action_handler.js [hunk 1]
```diff
@@ -212,38 +212,7 @@ const ActionHandler = Mixin.create({
...
-        'Specifying actions in `_actions` is deprecated, please use `actions` instead.',
-        false,
-        { id: 'ember-runtime.action-handler-_actions', until: '3.0.0' }
-      );
-
-      props.actions = props._actions;
-      delete props._actions;
-    }
-  }
+  willMergeMixin() {}
```
Comment (rwjblue) [inline packages/ember-runtime/lib/mixins/action_handler.js]: I think we can remove the implementation completely?

**Review by rwjblue — COMMENTED** [review]
Looking good! 

One inline tweak (we can remove the whole `willMergeMixin` function in the action handler mixin, and can you also add a comment above https://github.com/emberjs/ember.js/blob/d074b82205ad1c7b2e84e29cfc3d081f5cc2ae7c/packages/ember-metal/lib/mixin.js#L241

to make it clear that this can be removed after 3.4?

File: packages/ember-runtime/lib/mixins/action_handler.js (hunk 1 above)
Comment (thoov) [reply packages/ember-runtime/lib/mixins/action_handler.js]: Will this have issues with https://github.com/emberjs/data/blob/master/addon/-private/system/model/model.js#L1889

Comment (rwjblue) [reply packages/ember-runtime/lib/mixins/action_handler.js]: No, removing this empty function shouldn’t affect that (we aren’t removing the functionality yet)...

**Comment by thoov** [comment]
@rwjblue should be good now

**Comment by rwjblue** [comment]
Awesome, thank you!!

**Pull request merged**